docs/deployment/rds-s3/README.md

```diff
@@ -141,29 +141,135 @@ Resources:
+   1. [RDS] Configure `awsconfigs/apps/pipeline/rds/params.env` file with the RDS endpoint URL that was configured when following the steps in Create RDS Instance steps in prerequisites(#1-prerequisites).
 
-   1. Configure `awsconfigs/apps/pipeline/params.env` file with the RDS endpoint URL, S3 bucket name, and S3 bucket region that were configured when following the steps in Create RDS Instance and Create S3 Bucket steps in prerequisites(#1-prerequisites).
-
-      - For example, if your RDS endpoint URL is `rm12abc4krxxxxx.xxxxxxxxxxxx.us-west-2.rds.amazonaws.com`, S3 bucket name is `kf-aws-demo-bucket`, and s3 bucket region is `us-west-2` your `params.env` file should look like:
+      - For example, if your RDS endpoint URL is `rm12abc4krxxxxx.xxxxxxxxxxxx.us-west-2.rds.amazonaws.com` your `params.env` file should look like:
       - ```
         dbHost=rm12abc4krxxxxx.xxxxxxxxxxxx.us-west-2.rds.amazonaws.com
+        ```
 
+   2. [S3] Configure `awsconfigs/apps/pipeline/s3/params.env` file with the S3 bucket name, and S3 bucket region that were configured when following the steps in Create S3 Bucket steps in prerequisites(#1-prerequisites).
+
+      - For example, if your S3 bucket name is `kf-aws-demo-bucket` and s3 bucket region is `us-west-2` your `params.env` file should look like:
+      - ```
         bucketName=kf-aws-demo-bucket
         minioServiceHost=s3.amazonaws.com
         minioServiceRegion=us-west-2
         ```
 
 ### 4. Build Manifests and Install Kubeflow
 
+Choose your deployment option below from:
+- Deploying the configuration for both RDS and S3
+- Deploying the configuration for RDS only
+- Deploying the configuration for S3 only
+
+#### [RDS and S3] Deploy both RDS and S3
+
 ```sh
 while ! kustomize build docs/deployment/rds-s3 | kubectl apply -f -; do echo "Retrying to apply resources"; sleep 10; done
 ```
 
+#### [RDS] Deploy RDS only
+
+```sh
+while ! kustomize build docs/deployment/rds-s3/rds-only | kubectl apply -f -; do echo "Retrying to apply resources"; sleep 10; done
+```
+
+#### [S3] Deploy S3 only
+
+```sh
+while ! kustomize build docs/deployment/rds-s3/s3-only | kubectl apply -f -; do echo "Retrying to apply resources"; sleep 10; done
+```
+
 Once, everything is installed successfully, you can access the Kubeflow Central Dashboard [by logging in to your cluster](../vanilla/README.md#connect-to-your-kubeflow-cluster).
 
 Congratulations! You can now start experimenting and running your end-to-end ML workflows with Kubeflow.
 
+## Configuring KFP with only RDS or only S3
+
+### RDS Only
+
+Follow the [install](#install) steps prefixed with `[RDS]` as well those without any prefix.
+
+### S3 Only
+
+Follow the [install](#install) steps prefixed with `[S3]` as well those without any prefix.
+
+## Changing the RDS/S3 secret name
+
+### Changing the RDS secret name
+Update the `objectName` under `parameters.objects` in to the desired secret name in `awsconfigs/apps/pipeline/rds/secret-provider.yaml`
```

Comment:
Is this something kustomize can do by passing the value in values file?

Reply:
The problem is that `objectName` is part of a string literal that looks like:
```
  parameters:
    objects: | 
      - objectName: "rds-secret"
        objectType: "secretsmanager"
        jmesPath:
            - path: "username"
              objectAlias: "user"
            - path: "password"
              objectAlias: "pass"
            - path: "host"
              objectAlias: "host"
            - path: "database"
              objectAlias: "database"
            - path: "port"
              objectAlias: "port"
```

It is possible to patch `parameters.objects`, and the patch file would look like:

```
apiVersion: secrets-store.csi.x-k8s.io/v1alpha1
kind: SecretProviderClass
metadata:
  name: rds-secret
  namespace: kubeflow
spec:
  parameters:
    objects: | 
      - objectName: "another-rds-secret-name"   # changed this
        objectType: "secretsmanager"
        jmesPath:
            - path: "username"
              objectAlias: "user"
            - path: "password"
              objectAlias: "pass"
            - path: "host"
              objectAlias: "host"
            - path: "database"
              objectAlias: "database"
            - path: "port"
              objectAlias: "port"
```
but would need to include the other fields in the string literal that we don't want to replace (such as jmesPath), and would be error prone as well.

The other (hacky) approach would be to use [vars](https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/vars/), but by [default](https://github.com/kubernetes-sigs/kustomize/blob/master/api/konfig/builtinpluginconsts/varreference.go) kustomize doesn't support using vars to replace strings in custom resources, such as SecretProviderClass.

I'm also not able to find a way to convert a yaml file to string literal, so that we can patch the yaml and insert the patched yaml into the SecretProvider as a string literal.

So overall I was not able to find a kustomize solution that works with the format of the SecretProviderClass parameters.objects field.